default/AI/ProductionAI.py

```diff
@@ -77,101 +87,71 @@ def get_design_cost(cur_turn, design, pid):
     elif isinstance(loc, list):
-        planet_ids = loc
+        planet_ids = set(loc).intersection(ColonisationAI.empire_shipyards)
     elif isinstance(loc, int):
-        planet_ids = [loc]
+        if loc in ColonisationAI.empire_shipyards:
+            planet_ids = [loc]
+        else:
+            return None, None, None
     else:  # problem
         return None, None, None
-
-    if priority in design_cache:  # use new framework
+    if priority in design_cache:
         best_designs = design_cache[priority]
         if not best_designs:
             return None, None, None
-        top_rating, top_id = best_designs[0][0], best_designs[0][2]
+
+        for design_stats in best_designs:
+            top_rating, pid, top_id, cost = design_stats
+            if pid in planet_ids:
+                break
         valid_locs = [item[1] for item in best_designs if item[0] == top_rating and item[2] == top_id]
         return top_id, fo.getShipDesign(top_id), valid_locs
-    else:  # use old framework
-        empire = fo.getEmpire()
-        empire_id = empire.empireID
-        these_design_ids = []
-        design_name_bases = shipTypeMap.get(priority, ["nomatch"])
-        for base_name in design_name_bases:
-            these_design_ids.extend([(design_name_bases[base_name] + fo.getShipDesign(design).name(False), design)
-                                     for design in empire.availableShipDesigns
-                                     if base_name in fo.getShipDesign(design).name(False)])
-        if not these_design_ids:
-            return None, None, None  # must be missing a Shipyard (or checking for outpost ship but missing tech)
-        # ships = [(fo.getShipDesign(design).name(False), design) for design in these_design_ids]
-
-        for _, design_id in sorted(these_design_ids, reverse=True):
-            design = fo.getShipDesign(design_id)
-            valid_locs = []
-            for pid in planet_ids:
-                if pid is None:
-                    continue
-                if design.productionLocationForEmpire(empire_id, pid):
-                    valid_locs.append(pid)
-            if valid_locs:
-                return design_id, design, valid_locs
+    else:
         return None, None, None  # must be missing a Shipyard or other orbital (or missing tech)
 
 
-def getBestShipRatings(loc=None, verbose=False):
+def getBestShipRatings(loc=None):
     """returns list of [partition, pid, designID, design] sublists, currently only for military ships"""
+    # Since we haven't yet implemented a way to target military ship construction at/near particular locations
+    # where they are most in need, and also because our rating system is presumably useful-but-not-perfect, we want to
+    # distribute the construction across the Resource Group and across similarly rated designs, preferentially choosing
+    # the best rated design/loc combo, but if there are multiple design/loc combos with the same or similar ratings then
```

Comment:
Then I see `loc` I read it as `synchronization lock`. Every one in project come with own background. I suggest not to use `loc` and other abbreviations and shrinking in code.
	
Same with `pid` it will always be `Process IDentifier`  for me.